sympy/solvers/tests/test_solveset.py

```diff
@@ -1573,22 +1537,23 @@ def test_raise_exception_nonlinsolve():
     raises(NotImplementedError, lambda: nonlinsolve([(x+y)**2 - 9, x**2 - y**2 - 0.75], (x, y)))
 
 
-def test_trig_system():
+@XFAIL
+def test_trig_system_fail():
     # TODO: add more simple testcases when solveset returns
     # simplified soln for Trig eq
+
+    # fails because it gives Lambda(n, n*pi + pi/2) and
+    # 3*pi/2 does not satisfy first equation (giving -2 instead of 0)
     assert nonlinsolve([sin(x) - 1, cos(x) -1 ], x) == S.EmptySet
```

Comment:
this line can be moved out of the failing tests, I think

Reply:
IMHO, more important is to make `_solve_trig` smarter, afterward making the modifications mentioned above( https://github.com/sympy/sympy/pull/21196#issuecomment-814539934) than afterward these comments can be removed.

I think it's a bit long journey, so IMO let's concentrate on another simpler PRs, if you don't mind : )

Reply:
Agree. Simplification routines can be a rabbit's trail.